android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/ButtonFactory.java

```diff
@@ -0,0 +1,123 @@
+        button.setLayoutParams(layoutParams);
+        button.setTextSize(TypedValue.COMPLEX_UNIT_PX, context.getResources()
+                .getDimension(R.dimen.button_text_size));
+        button.setTextColor(context.getResources().getColor(android.R.color.white));
+        button.setHeight(context.getResources().getDimensionPixelSize(R.dimen.button_height));
+
+        String hint = jsonObject.optString("hint");
+        if (!TextUtils.isEmpty(hint)) {
+            button.setText(hint);
+        }
+
+        button.setId(ViewUtil.generateViewId());
+        canvasIds.put(button.getId());
+
+        button.setTag(R.id.key, jsonObject.getString("key"));
+        button.setTag(R.id.openmrs_entity_parent, openMrsEntityParent);
+        button.setTag(R.id.openmrs_entity, openMrsEntity);
+        button.setTag(R.id.openmrs_entity_id, openMrsEntityId);
+        button.setTag(R.id.type, jsonObject.getString("type"));
+        button.setTag(R.id.address, stepName + ":" + jsonObject.getString("key"));
+
+        if (jsonObject.has("read_only")) {
+            button.setEnabled(!jsonObject.getBoolean("read_only"));
+            button.setFocusable(!jsonObject.getBoolean("read_only"));
+        }
+
+        JSONObject action = jsonObject.optJSONObject("action");
+        if (action != null) {
+            jsonObject.put("value", "false");
+            jsonObject.getJSONObject("action").put("result", false);
+            final String behaviour = action.optString("behaviour");
+            button.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    try {
+                        String addressString = (String) v.getTag(R.id.address);
+                        if (!TextUtils.isEmpty(addressString)) {
+                            String[] address = addressString.split(":");
+                            JSONObject jsonObject = ((JsonApi) context)
+                                    .getObjectUsingAddress(address);
+                            jsonObject.getJSONObject("action").put("result", false);
+                            jsonObject.put("value", "false");
+                            switch (behaviour) {
+                                case BEHAVIOUR_FINISH_FORM:
+                                    jsonObject.getJSONObject("action").put("result", true);
+                                    jsonObject.put("value", "true");
+                                    formFragment.save();
+                                    break;
+                                case BEHAVIOUR_NEXT_STEP:
+                                    jsonObject.getJSONObject("action").put("result", true);
+                                    jsonObject.put("value", "true");
+                                    formFragment.next();
+                                    break;
+                            }
+                        }
+                    } catch (Exception e) {
```

Comment:
this throws a generic exception? Is there anything more specific to catch instead?